inject/src/main/java/io/micronaut/inject/annotation/DefaultAnnotationMetadata.java

```diff
@@ -1962,7 +1962,7 @@ private void putValues(String annotation, Map<CharSequence, Object> values, Map<
             }
         } else {
             if (!hasValues) {
-                existing = existing == null ? Collections.emptyMap() : existing;
+                existing = existing == null ? new LinkedHashMap<>() : existing;
```

Comment:
Could you alter this to `new LinkedHashMap<>(3)` to reduce memory consumption since most likely for the majority of cases this will remain empty